### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,35 +33,35 @@
   },
   "dependencies": {
     "broccoli-funnel": "^2.0.1",
-    "ember-cli-babel": "^6.12.0",
+    "ember-cli-babel": "^7.1.0",
```
Comment (rwjblue) [inline package.json]: Need to roll this back for now (ember-cli-babel@7 requires Node 6+)

File: package.json [hunk 2]
```diff
@@ -33,35 +33,35 @@
...
     "ember-cli-eslint": "^4.2.1",
-    "ember-cli-htmlbars": "^2.0.1",
+    "ember-cli-htmlbars": "^3.0.0",
     "ember-cli-inject-live-reload": "^1.4.1",
-    "ember-cli-pretender": "^1.0.1",
+    "ember-cli-pretender": "^3.0.0",
     "ember-cli-shims": "^1.2.0",
     "ember-cli-test-loader": "^2.2.0",
-    "ember-data": "~3.1.1",
+    "ember-data": "~3.4.0",
```
Comment (rwjblue) [inline package.json]: Change should be rolled back, ember-data latest requires Node 6

File: package.json [hunk 3]
```diff
@@ -33,35 +33,35 @@
...
-    "ember-source": "~3.1.0",
+    "ember-resolver": "^5.0.1",
+    "ember-source": "~3.4.0",
     "ember-source-channel-url": "^1.0.1",
     "ember-try": "^1.0.0-beta.1",
-    "eslint-config-prettier": "^2.6.0",
+    "eslint-config-prettier": "^3.0.1",
     "eslint-plugin-disable-features": "^0.1.3",
-    "eslint-plugin-node": "^6.0.1",
+    "eslint-plugin-node": "^7.0.1",
```
Comment (rwjblue) [inline package.json]: I think we need to roll this back until we drop Node 4

**Comment by wagenet** [comment]
Turns out that almost all the dependencies I upgraded depend on Node 6, so not sure if it makes sense to fix this PR. @rwjblue

**Comment by Turbo87** [comment]
@wagenet https://github.com/emberjs/ember-test-helpers/pull/427 dropped Node 4 support, so this PR could be revived

**Comment by rwjblue** [comment]
@wagenet - Looks like we only have a `yarn.lock` conflict now, mind rebasing and fixing that up?

File: .travis.yml [hunk 4]
```diff
@@ -72,7 +77,10 @@ before_install:
   - export PATH=$HOME/.yarn/bin:$PATH
 
 install:
-  - yarn install
+  - yarn install --no-lockfile --non-interactive
```
Comment (Turbo87) [inline .travis.yml]: This looks wrong to me

File: .travis.yml [hunk 5]
```diff
@@ -72,7 +77,10 @@ before_install:
...
 
 script:
+  - yarn lint:js
```
Comment (Turbo87) [inline .travis.yml]: We already do this in the "test" test

File: package.json [hunk 6]
```diff
@@ -21,14 +22,13 @@
...
-  "repository": "https://github.com/emberjs/ember-test-helpers",
   "scripts": {
     "build": "ember build",
     "changelog": "lerna-changelog",
     "docs": "scripts/precompile.sh && documentation build --document-exported \"addon-test-support/@ember/test-helpers/index.js\" --config documentation.yml -f md -o API.md && ember ts:clean",
-    "lint": "eslint *.js addon addon-test-support app config lib node-tests server tests test-support",
+    "lint": "eslint .",
     "start": "ember serve",
-    "test": "ember test",
+    "test": "EMBER_CLI_IGNORE_ADDON_NAME_MISMATCH=1 ember test",
```
Comment (Turbo87) [inline package.json]: is this the right fix? 🤔

**Comment by wagenet** [comment]
See #457 for some context on the failing tests.

File: package.json (hunk 6 above)
Comment (rwjblue) [reply package.json]: Ya, lets go ahead and change `index.js` to do the blueprint thing:

```json
  name: require('./package').name,
```

**Comment by wagenet** [comment]
@rwjblue so we're still blocked here because of the issue with `isSettled` :/

**Comment by rwjblue** [comment]
Lets skip that one test for now, then follow up on the `isSettled` thing in #457.

**Comment by wagenet** [comment]
Hmm... seeing a few of the scenarios failing. Will have to investigate further next week.

**Comment by wagenet** [comment]
@rwjblue finally passing!

File: config/ember-try.js [hunk 7]
```diff
@@ -8,11 +8,16 @@ module.exports = function() {
...
+    let env = {
+      EMBER_OPTIONAL_FEATURES: JSON.stringify({ 'jquery-integration': false }),
+    };
+
     return {
       useYarn: true,
       scenarios: [
         {
           name: 'ember-2.0',
+          env,
```
[the remark this answers is not in the document]
Comment (wagenet) [reply config/ember-try.js]: Hmm... well we _can_ use it, it just doesn't do anything meaningful, so I agree that it is confusing.

[the remark this answers is not in the document]
Comment (wagenet) [reply config/ember-try.js]: Which scenarios are actually expected to have jQuery?

**Comment by rwjblue** [comment]
Some conflicts now also (sorry for the delays / run around here)...

**Pull request merged**